grakn-graql/src/main/java/ai/grakn/graql/internal/parser/QueryVisitor.java

```diff
@@ -717,17 +730,6 @@ private Order getOrder(TerminalNode order) {
     }
 
     private ResourceType.DataType getDatatype(TerminalNode datatype) {
-        switch (datatype.getText()) {
-            case "long":
-                return ResourceType.DataType.LONG;
-            case "double":
-                return ResourceType.DataType.DOUBLE;
-            case "string":
-                return ResourceType.DataType.STRING;
-            case "boolean":
-                return ResourceType.DataType.BOOLEAN;
-            default:
-                throw new RuntimeException("Unrecognized datatype " + datatype.getText());
-        }
+        return QueryParser.DATA_TYPES.get(datatype.getText());
```

Comment:
It will return `null`! As long as the `DATA_TYPES` list is complete this will never happen.

I added the `throws` in the old method because there had to be a default case. I guess we could add `assert(result != null)` for the same effect here.